### ISSUE TITLE
lobby list spam exploit

### ISSUE THREAD

**Issue by GameRookies** [Tsuey/L4D2-Community-Update#456]
### Description

some L4D2 server owner through send fake lobby datapack(maybe speical UDP datapack?) to the master server, insert and spam their server's IP into the lobby list queue. when players choose quickmatch or lobby list in L4D2, they can only connect to the spam server, the other valve official server and community server are both hardly to be connected through the matchmaking system. The HACKER wants every player join his PAY-FOR-PLAY server to get money. It happens on all of the L4D2 master server in China.

Actually, I don't know how did they use this exploit. These servers have no lobby cookies, but player can still join them from lobby list and quickmatch. I hope the lobby serverside can improve some verification in order to prevent the vicious server spam their IP in the lobby list.

btw: sorry for my bad english.

### Reproduction steps

When player start game with "quickmatch" or "lobby list", the lobby queue is full with the simple spam server, valve official server and other community server are hardly to be connected. 

P.S: only China's masterserver fall into this sitiuation.

**Comment by Kaze1027** [comment]
Yes, it's still happening every seconds, someone come and help us.
![插入情况](https://github.com/Tsuey/L4D2-Community-Update/assets/51438620/0967d827-e6a0-46d2-be18-e66d8c1e886a)
![详情](https://github.com/Tsuey/L4D2-Community-Update/assets/51438620/9b85cc7f-3a43-42db-899e-c971d3ca39ee)

**Comment by Diver76** [comment]
similar problems :
https://github.com/ValveSoftware/Source-1-Games/issues/5297

**Comment by Diver76** [comment]
> Yes, it's still happening every seconds, someone come and help us. ![插入情况](https://private-user-images.githubusercontent.com/51438620/292396832-0967d827-e6a0-46d2-be18-e66d8c1e886a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQwOTU1NTQsIm5iZiI6MTcwNDA5NTI1NCwicGF0aCI6Ii81MTQzODYyMC8yOTIzOTY4MzItMDk2N2Q4MjctZTZhMC00NmQyLWJlMTgtZTY2ZDhjMWU4ODZhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAxVDA3NDczNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ1MjJlYjM0ZjhkOTFlZDUwNWU2YjNhNGM1YjNlN2EzNDQwNzE2ZjQ0MjZlNmI2NGQxZGM0NWFjZjY2OGRlYTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DqWwD50SzIf6Jn19KCLaM_BSZqsnmPndIEsItAmgG0Q) ![详情](https://private-user-images.githubusercontent.com/51438620/292396860-9b85cc7f-3a43-42db-899e-c971d3ca39ee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQwOTU1NTQsIm5iZiI6MTcwNDA5NTI1NCwicGF0aCI6Ii81MTQzODYyMC8yOTIzOTY4NjAtOWI4NWNjN2YtM2E0My00MmRiLTg5OWUtYzk3MWQzY2EzOWVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAxVDA3NDczNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5ZmU4YmMxNTEzMWI5NzlhMjM4NmQ4MzQ4YjRlZWFkN2JlZGY5Zjg3NmU3ZGUzM2E0YTA0Mjg5YTRiYjk3ZmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QcNTuAPoxXCcfAumlXkd-r8SoxjzPHWooR8f13watlo)

They just simply created many servers then edited the UDP packets sent to the clients to display fake player counts and names, they even edited the timestamps to ensure their servers always appeared at the top of the list. CSGO has also been plagued by similar problems for many years.
The reason these servers appear in your list is that they have associated themselves with your Steam group. Since 32-bit group ID can be easily derived from 64-bit ID, they can bind their servers to anyone's group. Valve has been aware of this issue for a long time, but they do not have any plans to fix it.
